pkg/bindings/images/images.go

```diff
@@ -0,0 +1,189 @@
+// use the optional size parameter.
+func GetImage(ctx context.Context, nameOrID string, size *bool) (*inspect.ImageData, error) {
+	conn, err := bindings.GetConnectionFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	params := make(map[string]string)
+	if size != nil {
+		params["size"] = strconv.FormatBool(*size)
+	}
+	inspectedData := inspect.ImageData{}
+	response, err := conn.NewRequest(http.MethodGet, fmt.Sprintf("/images/%s/json", nameOrID), nil, params)
+	if err != nil {
+		return &inspectedData, err
+	}
+	return &inspectedData, response.Process(&inspectedData)
+}
+
+func ImageTree(ctx context.Context, nameOrId string) error {
+	return bindings.ErrNotImplemented
+}
+
+// History returns the parent layers of an image.
+func History(ctx context.Context, nameOrID string) ([]handlers.HistoryResponse, error) {
+	var history []handlers.HistoryResponse
+	conn, err := bindings.GetConnectionFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	response, err := conn.NewRequest(http.MethodGet, fmt.Sprintf("/images/%s/history", nameOrID), nil, nil)
+	if err != nil {
+		return history, err
+	}
+	return history, response.Process(&history)
+}
+
+func Load(ctx context.Context, r io.Reader) error {
+	conn, err := bindings.GetConnectionFromContext(ctx)
+	if err != nil {
+		return err
+	}
+	// TODO this still needs error handling added
+	//_, err := http.Post(c.makeEndpoint("/images/loads"), "application/json", r) //nolint
+	_ = conn
+	return bindings.ErrNotImplemented
+}
+
+// Remove deletes an image from local storage.  The optional force parameter will forcibly remove
+// the image by removing all all containers, including those that are Running, first.
+func Remove(ctx context.Context, nameOrID string, force *bool) ([]map[string]string, error) {
+	var deletes []map[string]string
+	conn, err := bindings.GetConnectionFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	params := make(map[string]string)
+	if force != nil {
+		params["force"] = strconv.FormatBool(*force)
+	}
+	response, err := conn.NewRequest(http.MethodDelete, fmt.Sprintf("/images/%s", nameOrID), nil, params)
+	if err != nil {
+		return nil, err
+	}
+	return deletes, response.Process(&deletes)
+}
+
+// Export saves an image from local storage as a tarball or image archive.  The optional format
+// parameter is used to change the format of the output.
+func Export(ctx context.Context, nameOrID string, w io.Writer, format *string, compress *bool) error {
+	conn, err := bindings.GetConnectionFromContext(ctx)
+	if err != nil {
+		return err
+	}
+	params := make(map[string]string)
+	if format != nil {
+		params["format"] = *format
+	}
+	if compress != nil {
+		params["compress"] = strconv.FormatBool(*compress)
+	}
+	response, err := conn.NewRequest(http.MethodGet, fmt.Sprintf("/images/%s/get", nameOrID), nil, params)
+	if err != nil {
+		return err
+	}
+	if err := response.Process(nil); err != nil {
+		return err
+	}
+	_, err = io.Copy(w, response.Body)
```

Comment:
I believe you need a reponse.Body.Close() after the copy.

Reply:
discussed and resolved